From d432c47247a1325f898e526955fd172dc309a628 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 23 May 2020 11:20:55 +0200
Subject: [PATCH 1/1] doc: dfu: describe eMMC partition number parameter

In dfu_alt_info for eMMC the eMMC partition number can be specified.

The separator in dfu_alt_info is a semicolon not a comma.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 doc/README.dfu | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/doc/README.dfu b/doc/README.dfu
index caf1c9998c..29230ec654 100644
--- a/doc/README.dfu
+++ b/doc/README.dfu
@@ -49,7 +49,7 @@ Configuration Options:
 
 Environment variables:
   the dfu command use 3 environments variables:
-  "dfu_alt_info" : the DFU setting for the USB download gadget with a comma
+  "dfu_alt_info" : the DFU setting for the USB download gadget with a semicolon
                    separated string of information on each alternate:
                    dfu_alt_info="<alt1>;<alt2>;....;<altN>"
 
@@ -87,12 +87,21 @@ Commands:
   "mmc" (for eMMC and SD card)
     cmd: dfu 0 mmc <dev>
     each element in "dfu_alt_info" =
-      <name> raw <offset> <size>   raw access to mmc device
-      <name> part <dev> <part_id>  raw acces to partition
-      <name> fat <dev> <part_id>   file in FAT partition
-      <name> ext4 <dev> <part_id>  file in EXT4 partition
+      <name> raw <offset> <size> [mmcpart <num>]   raw access to mmc device
+      <name> part <dev> <part_id> [mmcpart <num>]  raw access to partition
+      <name> fat <dev> <part_id> [mmcpart <num>]   file in FAT partition
+      <name> ext4 <dev> <part_id> [mmcpart <num>]  file in EXT4 partition
+
+      with <partid> being the GPT or DOS partition index,
+      with <num> being the eMMC hardware partition number.
+
+    A value of environment variable dfu_alt_info for eMMC could be:
+
+      "u-boot raw 0x3e 0x800 mmcpart 1;bl2 raw 0x1e 0x1d mmcpart 1"
+
+    A value of environment variable dfu_alt_info for SD card could be:
 
-      with <partid> is the GPT or DOS partition index
+      "u-boot raw 0x80 0x800;uImage ext4 0 2"
 
   "nand" (raw slc nand device)
     cmd: dfu 0 nand <dev>
-- 
2.26.2

